google-ads-examples/src/main/java/com/google/ads/googleads/examples/remarketing/UpdateTargetingSetting.java

```diff
@@ -0,0 +1,191 @@
+        googleAdsClient.getLatestVersion().createGoogleAdsServiceClient()) {
+      // Creates a search request that retrieves the targeting settings from a given ad group.
+      String searchQuery =
+          "SELECT ad_group.id, ad_group.name, ad_group.targeting_setting.target_restrictions FROM"
+              + " ad_group WHERE ad_group.id = "
+              + adGroupId;
+
+      // Creates a request that will retrieve all ad groups using pages of the specified page size.
+      SearchGoogleAdsRequest request =
+          SearchGoogleAdsRequest.newBuilder()
+              .setCustomerId(Long.toString(customerId))
+              .setPageSize(PAGE_SIZE)
+              .setQuery(searchQuery)
+              .build();
+      // Issues the search request.
+      SearchPagedResponse searchPagedResponse = googleAdsServiceClient.search(request);
+      // Iterates over all rows in all pages and prints the requested field values for the ad group
+      // in each row.
+      for (GoogleAdsRow googleAdsRow : searchPagedResponse.iterateAll()) {
+        AdGroup adGroup = googleAdsRow.getAdGroup();
+        // Prints the results.
+        System.out.printf(
+            "Ad group with ID %d and name '%s' was found with the following targeting"
+                + " restrictions.%n",
+            adGroup.getId().getValue(), adGroup.getName().getValue());
+        List<TargetRestriction> targetRestrictions =
+            adGroup.getTargetingSetting().getTargetRestrictionsList();
+        // Loops through and prints each of the target restrictions.
+        // Reconstructs the TargetingSetting object with the updated audience target restriction
+        // because Google will overwrite the entire targeting_setting field of the ad group when
+        // the field mask includes targeting_setting in an update operation.
+        for (TargetRestriction targetRestriction : targetRestrictions) {
+          TargetingDimension targetingDimension = targetRestriction.getTargetingDimension();
+          boolean bidOnly = targetRestriction.getBidOnly().getValue();
+          System.out.printf(
+              "  Targeting restriction with targeting dimension '%s' and bid only set to '%b'.%n",
+              targetingDimension, bidOnly);
+          // Adds the target restriction to the TargetingSetting object as is if the targeting
+          // dimension has a value other than AUDIENCE because those should not change.
+          if (!targetingDimension.equals(TargetingDimension.AUDIENCE)) {
+            targetingSettingBuilder.addTargetRestrictions(targetRestriction);
+            // Adds the target restriction to the TargetingSetting object with bid_only set to true
+            // if the targeting dimension equals AUDIENCE in order to update that target
+            // restriction to "Observation".
+            // For more details about the targeting setting, visit
+            // https://support.google.com/google-ads/answer/7365594.
```

Comment:
This should be moved to right after "else".
I guess you may want to put this before "else", to be parallel to the way you do for "if", but it's quite confusing to see the comment, but nothing follows it before the end of the block.